streams/src/test/java/org/apache/kafka/streams/processor/internals/PartitionGroupTest.java

```diff
@@ -140,49 +140,47 @@ public void testTimeTracking() {
         record = group.nextRecord(info);
         // 1:[5, 2, 4]
         // 2:[4, 6]
-        // st: 3 (2's presence prevents it from advancing to 4)
         assertEquals(partition1, info.partition());
         assertEquals(3L, record.timestamp);
         assertEquals(5, group.numBuffered());
         assertEquals(3, group.numBuffered(partition1));
         assertEquals(2, group.numBuffered(partition2));
-        assertEquals(3L, group.timestamp());
+        assertEquals(4L, group.timestamp());
 
         // get one record, time should not be advanced
         record = group.nextRecord(info);
-        // 1:[2, 4]
-        // 2:[4, 6]
-        // st: 3 (2's presence prevents it from advancing to 4)
-        assertEquals(partition1, info.partition());
-        assertEquals(5L, record.timestamp);
+        // 1:[5, 2, 4]
+        // 2:[6]
+        assertEquals(partition2, info.partition());
+        assertEquals(4L, record.timestamp);
         assertEquals(4, group.numBuffered());
-        assertEquals(2, group.numBuffered(partition1));
-        assertEquals(2, group.numBuffered(partition2));
-        assertEquals(3L, group.timestamp());
+        assertEquals(3, group.numBuffered(partition1));
+        assertEquals(1, group.numBuffered(partition2));
+        assertEquals(5L, group.timestamp());
 
         // get one more record, now time should be advanced
         record = group.nextRecord(info);
-        // 1:[4]
-        // 2:[4, 6]
-        // st: 4
+        // 1:[2, 4]
+        // 2:[6]
+        // st: 5
         assertEquals(partition1, info.partition());
-        assertEquals(2L, record.timestamp);
+        assertEquals(5L, record.timestamp);
         assertEquals(3, group.numBuffered());
-        assertEquals(1, group.numBuffered(partition1));
-        assertEquals(2, group.numBuffered(partition2));
-        assertEquals(4L, group.timestamp());
+        assertEquals(2, group.numBuffered(partition1));
+        assertEquals(1, group.numBuffered(partition2));
+        assertEquals(5L, group.timestamp());
```

Comment:
Just a question: Why is this not `6L` ? (it should be `5L` after you applied the fix you want to do in a follow up PR).

Reply:
When we pop out 5, the stream time of partition1 is still 5 since the next record's ts is 2, and hence st stays as min(5, 6) = 5.